code/modules/projectiles/projectile/energy.dm

```diff
@@ -51,13 +51,23 @@
 	brightness = 15
 
 /obj/item/projectile/energy/flash/flare/on_impact(var/atom/A)
-	light_colour = pick("#e58775", "#ffffff", "#90ff90", "#a09030")
+	light_colour = pick("#e58775", "#ffffff", "#faa159", "#e34e0e")
 	set_light(1, 1, 4, 2, light_colour)
 	..() //initial flash
 
 	//residual illumination
 	new /obj/effect/effect/smoke/illumination(loc, rand(190,240), 8, 1, light_colour) //same lighting power as flare
 
+	var/turf/TO = get_turf(src)
+	var/area/AO = TO.loc
+	if(AO && (AO.area_flags & AREA_FLAG_EXTERNAL))
+		//Everyone saw that!
+		for(var/mob/living/mob in GLOB.living_mob_list_)
+			var/turf/T = get_turf(mob)
+			if(T && (T != TO) && (TO.z == T.z) && !mob.blinded)
```

Comment:
wouldn't this result in e.g. people seeing flashes from inside maintenance, since there's no external flag check on receiver end?

Reply:
I considered making a check for the other person, but I am not sure how to handle a person near a window not seeing shit either.

Reply:
Okay, pushed a solution. But maybe you have a better idea.